pkg/services/ngalert/api/api_provisioning.go

```diff
@@ -587,13 +588,18 @@ func exportResponse(c *contextmodel.ReqContext, body definitions.AlertingFileExp
 func exportHcl(download bool, body definitions.AlertingFileExport) response.Response {
 	resources := make([]hcl.Resource, 0, len(body.Groups)+len(body.ContactPoints)+len(body.Policies)+len(body.MuteTimings))
 	convertToResources := func() error {
-		for idx, group := range body.Groups {
+		for _, group := range body.Groups {
 			gr := group
-			resources = append(resources, hcl.Resource{
-				Type: "grafana_rule_group",
-				Name: fmt.Sprintf("rule_group_%04d", idx),
-				Body: &gr,
-			})
+			toHashedGruopIdx := append([]byte(gr.Name), []byte(gr.FolderUID)...)
+			hash, err := fnv.New64().Write(toHashedGruopIdx)
+			if err != nil {
+				resources = append(resources, hcl.Resource{
+					Type: "grafana_rule_group",
+					Name: fmt.Sprintf("rule_group_%016x", hash),
+					Body: &gr,
+				})
+			}
```

Comment:
Actually, this does not work. 
1. Method Write returns a number of bytes recorded and an error. It's not a hash. 
2. fnv implements, `io.Writer` interface, and never returns an error. So you can ignore it here.

The correct code could be 
```suggestion
           sum := fnv.New64()
           _, _ = sum.Write([]byte(gr.Name))
           _, _ = sum.Write([]byte(gr.FolderUID))
           hash := sum.Sum64()
           resources = append(resources, hcl.Resource{
					Type: "grafana_rule_group",
					Name: fmt.Sprintf("rule_group_%016x", hash),
					Body: &gr,
				})
```